pkg/util/kubernetes/apiserver/util.go

```diff
@@ -31,15 +32,36 @@ func SyncInformers(informers map[InformerName]cache.SharedInformer) error {
 	syncTimeout := config.Datadog.GetDuration("kube_cache_sync_timeout_seconds") * time.Second
 	for name := range informers {
 		name := name // https://golang.org/doc/faq#closures_and_goroutines
+		config := retry.Config{
+			Name:              string(name),
+			Strategy:          retry.Backoff,
+			InitialRetryDelay: syncTimeout,
+			MaxRetryDelay:     syncTimeout + time.Duration(60)*time.Second,
+		}
 		g.Go(func() error {
-			ctx, cancel := context.WithTimeout(context.Background(), syncTimeout)
-			defer cancel()
-			start := time.Now()
-			if !cache.WaitForCacheSync(ctx.Done(), informers[name].HasSynced) {
-				return fmt.Errorf("couldn't sync informer %s in %v", name, time.Now().Sub(start))
+			nextTry := config.InitialRetryDelay
+			lastTry := false
+			for {
+				ctx, cancel := context.WithTimeout(context.Background(), nextTry)
+				defer cancel()
+				start := time.Now()
+				if !cache.WaitForCacheSync(ctx.Done(), informers[name].HasSynced) {
+					log.Warnf("couldn't sync informer %s in %v", name, time.Now().Sub(start))
+					nextTry = nextTry + (1<<config.RetryCount)*time.Second
+					config.RetryCount++
+					log.Warnf("increase kube_cache_sync_timeout_seconds to %s", nextTry)
+					if lastTry {
+						return fmt.Errorf("couldn't sync informer %s in %v", name, time.Now().Sub(start))
+					}
+					if nextTry >= config.MaxRetryDelay {
+						nextTry = config.MaxRetryDelay
```

Comment:
what's the point of this line as it's already the case to enter the condition?

Reply:
It makes the `last retry time` not to be larger than the `max retry time`